### ISSUE TITLE
Error proxying to ingress 17 error: Error proxying request to origin: x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0

### ISSUE THREAD

**Issue by bdwilson** [cloudflare/cloudflared#362]
Attempting to proxy to my TP link Omada controller on 8043

```
# cloudflared --version
cloudflared version 2021.4.0 (built 2021-04-07-2048 UTC)
```

![image](https://user-images.githubusercontent.com/317946/115903916-27eb8a80-a432-11eb-9988-a0092b37e87a.png)

cloudflared tunnel configured as such:
```
  - hostname: omada.mydomain.us
    service: https://192.168.1.2:8043
    originRequest:
      noTLSverify: true
      originServerName: localhost
```
      

Error:
```
2021-04-23T16:44:59Z ERR CF-RAY: 64488f45dd620cbd-EWR Proxying to ingress 17 error: Error proxying request to origin: x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
2021-04-23T16:45:00Z ERR CF-RAY: 64488f4e89fe0cbd-EWR Proxying to ingress 17 error: Error proxying request to origin: x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
```

**Comment by bdwilson** [comment]
If I run Cloudflared with this env var `GODEBUG="x509ignoreCN=0" cloudflared --config=config.yml tunnel run` I then get this when going to a site with the above cert. 

`Proxying to ingress 17 error: Error proxying request to origin: x509: certificate signed by unknown authority`

Isn’t this the point of noTLSverify to not care about the authority?

**Comment by bdwilson** [comment]
I moved away from Omada but moved to Unifi.  I couldn't get this to work on Unifi Controller either, but I did find some guidance on getting a letsencrypt certificate into Unifi Controller (https://www.incredigeek.com/home/auto-renew-ssl-cert-with-unifi-running-in-docker). So now that my Unifi controller has a legit certificate (I used cloudflare certbot dns plugin in this script to register unifi.mydomain.us so the host didn't need to be on the internet), I was still unable to connect.  The solution that solved the issue is to do this in config.yml.

  - hostname: unifi.mydomain.us
    service: https://192.168.1.225:8443
    originRequest:
      originServerName: "unifi.mydomain.us"

But also pull down lets encrypt root & intermediate certs (ISRG1X and R3 from here https://letsencrypt.org/certificates/), put them into a bundle (le-chain.pem) and launch cloudflared as so:

SSL_CERT_FILE=le-chain.pem cloudflared --config=config.yml tunnel run

This allows cloudflared to connect to my LE-protected internal endpoint.
